pkg/apis/apps/deployable/v1/deployable_types.go

```diff
@@ -1,160 +0,0 @@
-}
-
-// DeployableSpec defines the desired state of Deployable.
-type DeployableSpec struct {
-	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-
-	Template     *runtime.RawExtension        `json:"template"`
-	Dependencies []Dependency                 `json:"dependencies,omitempty"`
-	Placement    *placementv1alpha1.Placement `json:"placement,omitempty"`
-	Overrides    []Overrides                  `json:"overrides,omitempty"`
-	Channels     []string                     `json:"channels,omitempty"`
-}
-
-// DeployablePhase indicate the phase of a deployable.
-type DeployablePhase string
-
-const (
-	// DeployableUnknown means not processed by controller yet.
-	DeployableUnknown DeployablePhase = ""
-	// DeployableDeployed means successfully deployed to remote clusters.
-	DeployableDeployed DeployablePhase = "Deployed"
-	// DeployableFailed means failed to apply the template into system.
-	DeployableFailed DeployablePhase = "Failed"
-	// DeployablePropagated means created other managed deployables.
-	DeployablePropagated DeployablePhase = "Propagated"
-)
-
-// ResourceUnitStatus aggregates status from target clusters.
-type ResourceUnitStatus struct {
-	Phase          DeployablePhase `json:"phase"`
-	Reason         string          `json:"reason,omitempty"`
-	Message        string          `json:"message,omitempty"`
-	LastUpdateTime *metav1.Time    `json:"lastUpdateTime,omitempty"`
-
-	ResourceStatus *runtime.RawExtension `json:"resourceStatus,omitempty"`
-}
-
-// DeployableStatus defines the observed state of Deployable.
-type DeployableStatus struct {
-	// INSERT ADDITIONAL STATUS FIELD - define observed state of cluster
-	// Important: Run "make" to regenerate code after modifying this file
-	ResourceUnitStatus `json:",inline"`
-	PropagatedStatus   map[string]*ResourceUnitStatus `json:"targetClusters,omitempty"`
-}
-
-// +genclient
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
-// Deployable is the Schema for the deployables API
-// +k8s:openapi-gen=true
-// +kubebuilder:subresource:status
-// +kubebuilder:printcolumn:name="template-kind",type="string",JSONPath=".spec.template.kind",description="Kind of the template"
-// +kubebuilder:printcolumn:name="template-apiversion",type="string",JSONPath=".spec.template.apiVersion",description="api version of the template"
-// +kubebuilder:printcolumn:name="age",type="date",JSONPath=".metadata.creationTimestamp"
-// +kubebuilder:printcolumn:name="status",type="string",JSONPath=".status.phase"
-type Deployable struct {
-	metav1.TypeMeta   `json:",inline"`
-	metav1.ObjectMeta `json:"metadata,omitempty"`
-
-	Spec   DeployableSpec   `json:"spec"`
-	Status DeployableStatus `json:"status,omitempty"`
-}
-
-// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-
-// DeployableList contains a list of Deployable.
-type DeployableList struct {
```

Comment:
I guess it's no longer used?

Reply:
From what I can find yes, it is no longer being used. As well in the `hack/test` directory, deployable has no reference to a CRD which is used for writing tests for the other types like helmrelease and placementrule.

Reply:
Great, percentage pump to go!!

Reply:
Yeah, the deployable API has deprecated since 2.5. For now We just hold the deployable CRD for upgrade path.